hls-plugin-api/src/Ide/Types.hs

```diff
@@ -383,168 +406,201 @@ class HasTracing (MessageParams m) => PluginMethod (k :: MessageKind) (m :: Meth
-    where
-      uri = msgParams ^. L.textDocument . L.uri
+  pluginEnabled = pluginEnabledOnlyFileType
+
 
 instance PluginMethod Request Method_TextDocumentDocumentHighlight where
-  pluginEnabled _ msgParams pluginDesc _ =
-    pluginResponsible uri pluginDesc
-    where
-      uri = msgParams ^. L.textDocument . L.uri
+  pluginEnabled = pluginEnabledOnlyFileType
+
 
 instance PluginMethod Request Method_TextDocumentReferences where
-  pluginEnabled _ msgParams pluginDesc _ =
-    pluginResponsible uri pluginDesc
-    where
-      uri = msgParams ^. L.textDocument . L.uri
+  pluginEnabled = pluginEnabledOnlyFileType
 
 instance PluginMethod Request Method_WorkspaceSymbol where
   -- Unconditionally enabled, but should it really be?
-  pluginEnabled _ _ _ _ = True
+  pluginEnabled _ _ _ _ = PluginEnabled
 
 instance PluginMethod Request Method_TextDocumentCodeLens where
-  pluginEnabled _ msgParams pluginDesc config = pluginResponsible uri pluginDesc
-      && pluginEnabledConfig plcCodeLensOn (configForPlugin config pluginDesc)
-    where
-      uri = msgParams ^. L.textDocument . L.uri
+  pluginEnabled = pluginEnabledWithFeature plcCodeLensOn
 
 instance PluginMethod Request Method_CodeLensResolve where
   -- See Note [Resolve in PluginHandlers]
-  pluginEnabled _ msgParams pluginDesc config =
-    pluginResolverResponsible (msgParams ^. L.data_) pluginDesc
-    && pluginEnabledConfig plcCodeActionsOn (configForPlugin config pluginDesc)
+  pluginEnabled = pluginEnabledResolve plcCodeLensOn
 
 instance PluginMethod Request Method_TextDocumentRename where
-  pluginEnabled _ msgParams pluginDesc config = pluginResponsible uri pluginDesc
-      && pluginEnabledConfig plcRenameOn (configForPlugin config pluginDesc)
-   where
-      uri = msgParams ^. L.textDocument . L.uri
+  pluginEnabled = pluginEnabledWithFeature plcRenameOn
+
 instance PluginMethod Request Method_TextDocumentHover where
-  pluginEnabled _ msgParams pluginDesc config = pluginResponsible uri pluginDesc
-      && pluginEnabledConfig plcHoverOn (configForPlugin config pluginDesc)
-   where
-      uri = msgParams ^. L.textDocument . L.uri
+
+  pluginEnabled = pluginEnabledWithFeature plcHoverOn
 
 instance PluginMethod Request Method_TextDocumentDocumentSymbol where
-  pluginEnabled _ msgParams pluginDesc config = pluginResponsible uri pluginDesc
-      && pluginEnabledConfig plcSymbolsOn (configForPlugin config pluginDesc)
-    where
-      uri = msgParams ^. L.textDocument . L.uri
+
+  pluginEnabled = pluginEnabledWithFeature plcSymbolsOn
 
 instance PluginMethod Request Method_CompletionItemResolve where
   -- See Note [Resolve in PluginHandlers]
-  pluginEnabled _ msgParams pluginDesc config = pluginResolverResponsible (msgParams ^. L.data_) pluginDesc
-    && pluginEnabledConfig plcCompletionOn (configForPlugin config pluginDesc)
+  pluginEnabled _ (CompletionItem {_data_=Nothing}) (PluginDescriptor {pluginId=PluginId "ghcide-completions-bounce"}) _ = PluginEnabled
```

Comment:
This seems odd?

Reply:
Yes, thanks for catching this. It is leftover from the bounce stuff. Cleaned it up!